File: C#/6/6_3.cs
using System;

void SendEmail(string to, string subject, string body)
{
    Email email = new Email
    {
        To = to,
        Subject = subject,
        Body = body,
        SentAt = DateTime.Now,
        Status = "sent"
    };
    LogEmail(email);
}

for (int i = 0; i < 2; i++)
{
    Email welcomeEmail = new Email();
    SendEmail("newuser@example.com", "Welcome!", "Thank you for joining.");
    welcomeEmail.IsWelcomeEmail = true;
    LogEmail(welcomeEmail);
}

File: JavaScript/6/6_3.js
function sendEmail(to, subject, body) {
  let email = {
    to,
    subject,
    body,
    sentAt: new Date(),
    status: "sent",
  };
  logEmail(email);
  return email;
}

// Send a welcome email with an additional property
let welcomeEmail = sendEmail(
  "newuser@example.com",
  "Welcome!",
  "Thank you for joining.",
  { isWelcomeEmail: true }
);

// Send a password reset email with an additional property
let passwordResetEmail = sendEmail(
  "user@example.com",
  "Reset Password",
  "Click here to reset your password.",
  { isPasswordReset: true }
);
